measurement/plugins/netflix_fast/measurements.py

```diff
@@ -0,0 +1,334 @@
+    - `max_time` (presently 30s) expired.
+    - Results have become "stabilised"
+    - All threads have finished downloading
+    - A single thread has finished downloading, IF `terminate_on_thread_complete=True`
+
+Stabilisation is considered to be:
+    - Downloaded has been running longer than `MIN_TIME` (presently 3s)
+    - AND more than or equal to `STABLE_MEASUREMENTS_LENGTH` (presently 6) have been recorded
+    - AND maximum percentage delta in these measurements is `< STABLE_MEASUREMENTS_DELTA` presently (2%)
+
+In cases where the test concludes independently of the main loop (i.e when `reason_terminated == "thread_complete"`) The speed at the instant the thread completes is used, otherwise the final speed is used.
+
+All this is then packaged into a `NetflixFastMeasurementResult`
+
+After this, each of the URLs downloaded from has a latency test then have an Honesty-Box LatencyMeasurement test run against them, the results of which, along with the location and download rates/sizes for each thread are put into a `NetflixFastThreadResult`.
+
+All these results are then returned as a list.
+"""
+
+import requests
+import re
+import time
+import urllib
+import json
+from threading import Thread
+from collections import deque
+from statistics import mean
+
+from measurement.measurements import BaseMeasurement
+from measurement.results import Error
+from measurement.units import RatioUnit, TimeUnit, StorageUnit, NetworkUnit
+from measurement.plugins.latency.measurements import LatencyMeasurement
+from measurement.plugins.download_speed.measurements import DownloadSpeedMeasurement
+from measurement.plugins.netflix_fast.results import (
+    NetflixFastMeasurementResult,
+    NetflixFastThreadResult,
+)
+
+
+NETFLIX_ERRORS = {
+    "netflix-err": "Netflix test encountered an unknown error",
+    "netflix-ping": "Netflix test encountered an error when pinging hosts",
+    "netflix-response": "Netflix test received an invalid response from fast.com",
+    "netflix-script-regex": "Netflix test failed to find script in the response",
+    "netflix-script-response": "Netflix test received an invalid response from script",
+    "netflix-token-regex": "Netflix test failed to find token in the response",
+    "netflix-api-response": "Netflix test received an invalid response when querying for URLs",
+    "netflix-api-json": "Netflix test failed to decode URLs",
+    "netflix-api-parse": "Netflix test failed interpret elements of the decoded JSON",
+    "netflix-connection": "Netflix test failed to connect to download URLs",
+    "netflix-download": "Netflix test encountered an error downloading data",
+}
+CHUNK_SIZE = 64 * 2 ** 10
+MIN_TIME = 3
+PING_COUNT = 4
+STABLE_MEASUREMENTS_LENGTH = 6
```

Comment:
Is this `MEASUREMENTS_COUNTED_BEFORE_CONSIDERED_STABLE`?

Reply:
As @sjdines is demonstrating with this and the time-related variables, it's better to be explicit than implicit when naming things :-)

I always try to put myself in the position of debugging the code and coming across this variable having not dealt with it before. `STABLE_MEASUREMENTS_LENGTH` kinda hints at what this variable represents, but it would probably require me to do further digging to confirm. `MEASUREMENTS_COUNTED_BEFORE_CONSIDERED_STABLE` leaves little room for doubt.

Likewise `MIN_TIME` _probably_ means `MIN_TIME_SEC`. But just using `MIN_TIME_SEC` puts it beyond doubt...;-)